docs/architecture/ADR-005-QGB_posting_confirms_in_reserved_namespace.md

```diff
@@ -0,0 +1,96 @@
+# ADR 5: Using block space instead of state to post attestations confirms
+
+## Changelog
+
+- {date}: {changelog}
+
+## Context
+
+Currently, the QGB is using the state solely to save all the attestations and confirms it needs.
+We propose to post `Valset Confirms` and `DataCommitment Confirms` in a reserved namespace instead of adding them to the state to achieve the following:
+
+- Reduce the state machine complexity
+- Reduce the amount of state used by the QGB
+- Prepare for a more Rollup oriented QGB design
+
+## Alternative Approaches
+
+### Keep the existing design
+
+Keeping the current design would entail using the state extensively.
+This proves bad when the state grows after a few hundred attestations, and performing checks on the `Valset Confirms` and `DataCommitment Confirms` becomes so expensive.
+An example of such issue in here: [QGB data commitments/valsets state machine checks #631](https://github.com/celestiaorg/celestia-app/issues/631) and [Investigate the QGB transactions cost #603](https://github.com/celestiaorg/celestia-app/issues/603).
+
+Also, we will be forced to prune the state after the unbonding period not to end up with a gigantic state, issue defining this: [Prune the QGB state after the unbonding period ends #309](https://github.com/celestiaorg/celestia-app/issues/309).
```

Comment:
[question] why is it problematic to prune the state? I assume that the state that gets pruned is no longer necessary.

Reply:
No longer necessary is relative. In this case, pruning the state means that newly deployed QGB no  longer necessary contracts will not be able to sync from genesis. Only, from the current unbonding period. If we don't really care about that, then yes, makes sense to  prune.

Reply:
> No longer necessary is relative.

Sorry I don't follow. Are there scenarios where contracts on Ethereum would need the state from prior to the unbonding period (in other words the state we're considering pruning)?

Reply:
Sorry, just seen this message. What I meant by relative is:
- If we want any QGB contract to be able to sync from Genesis, then, we can't prune.
- If we want QGB contracts to only start from the current unbonding period. Then, yes, the old state is not necessary to keep.

Reply:
You explained this well, and I see the difference. I think we need to determine the intended use cases for QGB so that we can make this decision. In other words, we need to understand if QGB contracts would ever want to sync all the state from genesis. Since this is more of a product requirements question, I think it can be a separate issue and doesn't necessarily need to be resolved in this ADR.

Reply:
issue: https://github.com/celestiaorg/celestia-app/issues/683